roles/openshift_aws/tasks/build_elb_dict.yml

```diff
@@ -0,0 +1,12 @@
+---
+- name: querying elb
+  ec2_elb_facts:
+    names: "{{ openshift_aws_elb_names | join(',') }}"
+    region: "{{ openshift_aws_region }}"
+  register: elb_res
+
+# lib_utils_oo_list_of_dict_to_dict_from_key is a custom filter in
+# roles/lib_utils/filters/oo_filters.py
+- name: set elb fact dictionary
+  set_fact:
+    l_openshift_aws_elb_facts: "{{ elb_res | lib_utils_oo_list_of_dict_to_dict_from_key('name')}}"
```

Comment:
lib_utils_oo_list_of_dict_to_dict_from_key python function wasn't necessary before.  Why is this better?

Reply:
Previously (your patch set), we had to query ec2 for every record to resolve elb_name to elb_dnsname.  That is not great.

This builds a dictionary where we can lookup elb_facts based on elb_name in later tasks or plays.

Reply:
Also, the elb facts are not tied to any specific provider, so if we implement providers later, or someone utilizes their custom provider role, they can consume this data.